lib/core/src/putUtil.cpp

```diff
@@ -665,17 +699,26 @@ int
 putDirUtil( rcComm_t **myConn, char *srcDir, char *targColl,
             rodsEnv *myRodsEnv, rodsArguments_t *rodsArgs, dataObjInp_t *dataObjOprInp,
             bulkOprInp_t *bulkOprInp, rodsRestart_t *rodsRestart,
-            bulkOprInfo_t *bulkOprInfo ) {
+            bulkOprInfo_t *bulkOprInfo )
+{
+    namespace fs = boost::filesystem;
+
     char srcChildPath[MAX_NAME_LEN], targChildPath[MAX_NAME_LEN];
+    int status = 0;
 
     if ( srcDir == NULL || targColl == NULL ) {
```

Comment:
About the scope of "int status" in putUtil.cpp: although it is true that in general, it is a "best practice" to place the declaration as close as possible to where an object is used, not only in the sense of close proximity, but also in terms of the scope, there are exceptions.  

The way we use "int status" in the code everywhere, is as the object which carries the return from a prior function call.  It is used as the carrier of return information throughout the file, and what I want to avoid, is a maintainer of the code having to hunt down where the object is declared,
or worse used some other variable (int status2; ??) to hold status.  

Right at the beginning of the function is where it belongs for reasons of common sense.